packages/template-retail-react-app/app/commerce-api/index.js

```diff
@@ -112,72 +113,91 @@ class CommerceAPI {
+                            : !!sendCurrency
 
-                                const includeGlobalLocale = Array.isArray(sendLocale)
-                                    ? sendLocale.includes(prop)
-                                    : !!sendLocale
+                        fetchOptions['parameters'] = {
+                            ...(includeGlobalLocale ? {locale} : {}),
+                            ...(includeGlobalCurrency ? {currency} : {}),
+                            ...fetchOptions?.parameters
+                        }
+
+                        if (self.auth.pendingToken) {
+                            await self.auth.pendingToken
+                        }
 
-                                const includeGlobalCurrency = Array.isArray(sendCurrency)
-                                    ? sendCurrency.includes(prop)
-                                    : !!sendCurrency
+                        // Handle auth logic (replacing willSendRequest functionality)
+                        let dwsidHeader = {}
+                        const dwsid = self.auth.get('dwsid')
+                        if (dwsid) {
+                            dwsidHeader = {
+                                [DWSID_HEADER_KEY]: dwsid
+                            }
+                        }
 
-                                fetchOptions.parameters = {
-                                    ...(includeGlobalLocale ? {locale} : {}),
-                                    ...(includeGlobalCurrency ? {currency} : {}),
-                                    // Allowing individual API calls to override the global locale/currency
-                                    ...fetchOptions.parameters
-                                }
+                        // Special handling for auth methods
+                        if (
+                            methodName === 'authenticateCustomer' ||
+                            methodName === 'authorizeCustomer' ||
+                            methodName === 'getAccessToken'
+                        ) {
+                            return {
+                                ...options.parameters,
+                                headers: {
+                                    ...options.headers,
+                                    ...fetchOptions.headers
+                                },
+                                credentials: 'same-origin', // Required for SLAS calls to set dwsid cookie
+                                ...fetchOptions
+                            }
+                        }
 
-                                return self.willSendRequest(prop, ...args).then((newArgs) => {
-                                    return obj[prop](...newArgs).then((res) =>
-                                        self.didReceiveResponse(res, newArgs)
-                                    )
-                                })
+                        const {access_token: token} = await self.auth.ready()
```

Comment:
Should this be `self.auth.pendingToken`?

Reply:
So, CommerceAPI class gets instantiated in AppConfig.restore() which is one of the first things that runs when the app loads both on server and client.

So here we need to call `await self.auth.ready()` to trigger auth initialization instead of awaitng self.auth.pendingToken.

Also `self.auth.ready()` will await till auth calls are completed and we have an access_token available.

All other places in the app can then simply await `self.auth.pendingToken` which will resolve as soon as auth is setup.

Reply:
🤔 @shethj what if we call `self.auth.ready()` everywhere then? I feel like we shouldn't rely on `pendingToken`, as that's supposed to be private variable.

Reply:
Turns out using `auth.ready` behaves similar to pendingToken. I've replaced all occurrences of pending token to use auth.ready().

Thanks!